erpnext/hr/doctype/leave_application/leave_application.py

```diff
@@ -85,7 +84,7 @@ def validate_applicable_after(self):
 
 	def validate_dates(self):
 		if frappe.db.get_single_value("HR Settings", "restrict_backdated_leave_application"):
-			if self.from_date and self.from_date < frappe.utils.today():
+			if self.from_date and getdate(self.from_date) < getdate(today()):
```

Comment:
```suggestion
			if self.from_date and getdate(self.from_date) < getdate():
```
If you are using `getdate()` you don't need `today()` since getdate by default returns current date.